### ISSUE TITLE
DiagrammedAssertions can't cope with BigIntegers

### ISSUE THREAD

**Issue by pniederw** [scalatest/scalatest#611]
`assert(BigInteger.valueOf(1) == BigInteger.valueOf(2))`

```
Error:(8, 11) exception during macro expansion: 
scala.reflect.macros.TypecheckException: inferred type arguments [java.math.BigInteger.type] do not conform to method simpleExpr's type parameter bounds [T]
	at scala.reflect.macros.contexts.Typers$$anonfun$typecheck$2$$anonfun$apply$1.apply(Typers.scala:34)
	at scala.reflect.macros.contexts.Typers$$anonfun$typecheck$2$$anonfun$apply$1.apply(Typers.scala:28)
	at scala.reflect.macros.contexts.Typers$$anonfun$3.apply(Typers.scala:24)
	at scala.reflect.macros.contexts.Typers$$anonfun$3.apply(Typers.scala:24)
	at scala.reflect.macros.contexts.Typers$$anonfun$withContext$1$1.apply(Typers.scala:25)
	at scala.reflect.macros.contexts.Typers$$anonfun$withContext$1$1.apply(Typers.scala:25)
	at scala.reflect.macros.contexts.Typers$$anonfun$1.apply(Typers.scala:23)
	at scala.reflect.macros.contexts.Typers$$anonfun$1.apply(Typers.scala:23)
	at scala.reflect.macros.contexts.Typers$class.withContext$1(Typers.scala:25)
	at scala.reflect.macros.contexts.Typers$$anonfun$typecheck$2.apply(Typers.scala:28)
	at scala.reflect.macros.contexts.Typers$$anonfun$typecheck$2.apply(Typers.scala:28)
	at scala.reflect.internal.Trees$class.wrappingIntoTerm(Trees.scala:1716)
	at scala.reflect.internal.SymbolTable.wrappingIntoTerm(SymbolTable.scala:16)
	at scala.reflect.macros.contexts.Typers$class.withWrapping$1(Typers.scala:26)
	at scala.reflect.macros.contexts.Typers$class.typecheck(Typers.scala:28)
	at scala.reflect.macros.contexts.Context.typecheck(Context.scala:6)
	at scala.reflect.macros.contexts.Context.typecheck(Context.scala:6)
	at scala.reflect.macros.Typers$class.typeCheck(Typers.scala:58)
	at scala.reflect.macros.contexts.Context.typeCheck(Context.scala:6)
	at org.scalactic.MacroOwnerRepair.repairOwners(MacroOwnerRepair.scala:34)
	at org.scalatest.DiagrammedExprMacro.genMacro(DiagrammedExprMacro.scala:411)
	at org.scalatest.DiagrammedAssertionsMacro$.macroImpl(DiagrammedAssertionsMacro.scala:79)
	at org.scalatest.DiagrammedAssertionsMacro$.assert(DiagrammedAssertionsMacro.scala:85)
   assert(BigInteger.valueOf(1) == BigInteger.valueOf(2))
```

scala 2.11.6, scalatest 2.11:2.2.5

**Comment by cheeseng** [comment]
@pniederw the fix for this is already available: 

https://github.com/scalatest/scalatest/pull/583

but unfortunately it is in 2.3.x branch (to be released as 3.0.0) and not backported to 2.2.x branch yet, for workaround in 2.2.5 now you probably need to do: 

val a = BigInteger.valueOf(1)
val b = BigInteger.valueOf(2)
assert(a == b)

Sorry for the inconvenient caused.

**Comment by bvenners** [comment]
This was fixed and released last year, but we never closed the ticket. Closing now.
